core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -221,8 +213,6 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
   def quotaConfig: QuotaConfig = _quotaConfig
 
   /** ********* General Configuration ***********/
-  val brokerIdGenerationEnable: Boolean = getBoolean(ServerConfigs.BROKER_ID_GENERATION_ENABLE_CONFIG)
-  val maxReservedBrokerId: Int = getInt(ServerConfigs.RESERVED_BROKER_MAX_ID_CONFIG)
   var brokerId: Int = getInt(ServerConfigs.BROKER_ID_CONFIG)
```

Comment:
Pull request #18478 removes the `broker.id` configuration, but this pull request does not. Should we continue treating `broker.id` as synonymous with node.id in Kafka version 4.x? The [documentation](https://kafka.apache.org/documentation/#brokerconfigs_node.id) states that `node.id` is mandatory when running in KRaft mode. Therefore, it's likely safe to remove `broker.id` from version 4.x, as Kafka servers in KRaft mode should already be configured with `node.id`.

Reply:
If the behavior in 3.x is that the broker fails to start in KRaft mode unless node.id is set, then it would be fine to remove broker.id. But we should double check to make sure we didn't make an incompatible change in 4.0/trunk.

Reply:
> If the behavior in 3.x is that the broker fails to start in KRaft mode unless node.id is set

There's a potential regression if we remove `broker.id` because users can currently set it instead of `node.id`. However, this conflicts with our documentation, which states that `node.id` is mandatory. To address this, we could deprecate `broker.id` in version 4.x and then remove it entirely in version 5.x.

Reply:
ping @dajac if the approach of deprecating `broker.id` is acceptable.

Reply:
Deprecating `broker.id` has to be done via a KIP. Since there is no rush for that one, let's focus on making sure we don't break the behavior from 3.x for now.

Reply:
@FrankYang0529 Let's retain broker.id in issue #18478. Additionally, we should update the documentation to clarify that node.id and broker.id are synonymous. This will help prevent user confusion regarding their relationship.

Reply:
Thanks for the confirmation. I will update my PR.

Reply:
@ijuma @chia7712 do you think that we should add a warning message when users use `broker.id`? To let them know the configuration will be removed in 5.0.

Reply:
> do you think that we should add a warning message when users use broker.id? To let them know the configuration will be removed in 5.0.

`broker.id` is not deprecated in 4.0 (as @ijuma mentioned it needs a KIP), so the warning message is unnecessary to me. I open https://issues.apache.org/jira/browse/KAFKA-18628 to discuss the `broker.id`

Reply:
Yeah, we should discuss the node.id / broker.id thing in a separate JIRA or PR, I think.